client/my-sites/email/email-management/home/email-mailbox-warnings.jsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@automattic/components';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS,
+	EMAIL_WARNING_SLUG_UNVERIFIED_FORWARD,
+} from 'calypso/lib/emails/email-provider-constants';
+import {
+	getEmailForwardAddress,
+	hasGoogleAccountTOSWarning,
+	isEmailUserAdmin,
+	isGoogleEmailAccount,
+} from 'calypso/lib/emails';
+import { getGmailUrl, getGoogleAdminWithTosUrl } from 'calypso/lib/gsuite';
+import Gridicon from 'calypso/components/gridicon';
+import { isEmailForwardAccount } from 'calypso/lib/emails/is-email-forward-account';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { resendVerificationEmail } from 'calypso/state/email-forwarding/actions';
+
+const EmailMailboxWarningText = ( { text } ) => {
+	return (
+		<div className="email-mailbox-warnings__warning">
+			<Gridicon icon="info-outline" size={ 18 } />
+			<span>{ text }</span>
+		</div>
+	);
+};
+
+const EmailMailboxWarningAction = ( { buttonText, isExternal, ...otherProps } ) => {
+	return (
+		<div className="email-mailbox-warnings__action">
+			<Button compact { ...otherProps }>
+				<span>{ buttonText }</span>
+				{ isExternal && <Gridicon icon="external" /> }
+			</Button>
+		</div>
+	);
+};
+
+const resendEmailForwardVerification = ( mailbox, dispatch ) => {
+	const destination = getEmailForwardAddress( mailbox );
+	dispatch(
+		recordTracksEvent(
+			'calypso_email_management_email_forwarding_resend_verification_email_click',
+			{
+				destination,
+				domain_name: mailbox.domain,
+				mailbox: mailbox.mailbox,
+			}
+		)
+	);
+	dispatch( resendVerificationEmail( mailbox.domain, mailbox.mailbox, destination ) );
+};
+
+const getDetailsForWarning = ( { account, dispatch, mailbox, translate, warning } ) => {
+	const warningSlug = warning.warning_slug;
+
+	if ( isGoogleEmailAccount( account ) ) {
+		if ( warningSlug === EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS ) {
+			const emailAddress = `${ mailbox.mailbox }@${ mailbox.domain }`;
+
+			// If the the account-level Terms of Service need to be accepted, only admins can fix that
+			// issue, so ensure that we show the correct link for those users.
+			// For non-admins, not much can be done until the account-level Terms have been accepted.
+			if ( hasGoogleAccountTOSWarning( account ) ) {
+				if ( ! isEmailUserAdmin( mailbox ) ) {
+					return {
+						warningText: translate( 'Action required' ),
+					};
+				}
+
+				return {
+					actionProps: {
+						buttonText: translate( 'Finish setup' ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **34** times:
`translate( 'Finish Setup' )` *ES Score: 12*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/google-mailbox-warnings-to-email-plan-page)